x-pack/plugins/ml/public/application/components/model_snapshots/revert_model_snapshot_flyout/create_calendar.tsx

```diff
@@ -0,0 +1,289 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { FC, memo } from 'react';
+import { i18n } from '@kbn/i18n';
+import { FormattedMessage } from '@kbn/i18n/react';
+import moment from 'moment';
+// @ts-ignore
+import { formatDate } from '@elastic/eui/lib/services/format';
+import { XYBrushArea } from '@elastic/charts';
+import {
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiSpacer,
+  EuiFormRow,
+  EuiFieldText,
+  EuiDatePicker,
+  EuiButtonIcon,
+  EuiPanel,
+} from '@elastic/eui';
+
+import { EventRateChart } from '../../../jobs/new_job/pages/components/charts/event_rate_chart/event_rate_chart';
+import { Anomaly } from '../../../jobs/new_job/common/results_loader/results_loader';
+import { useCurrentEuiTheme } from '../../../components/color_range_legend';
+import { LineChartPoint } from '../../../jobs/new_job/common/chart_loader/chart_loader';
+
+export interface CalendarEvent {
+  start: moment.Moment | null;
+  end: moment.Moment | null;
+  description: string;
+}
+
+interface Props {
+  calendarEvents: CalendarEvent[];
+  setCalendarEvents: (calendars: CalendarEvent[]) => void;
+  minSelectableTimeStamp: number;
+  maxSelectableTimeStamp: number;
+  eventRateData: LineChartPoint[];
+  anomalies: Anomaly[];
+  chartReady: boolean;
+}
+
+export const CreateCalendar: FC<Props> = ({
+  calendarEvents,
+  setCalendarEvents,
+  minSelectableTimeStamp,
+  maxSelectableTimeStamp,
+  eventRateData,
+  anomalies,
+  chartReady,
+}) => {
+  const maxSelectableTimeMoment = moment(maxSelectableTimeStamp);
+  const minSelectableTimeMoment = moment(minSelectableTimeStamp);
+
+  const { euiTheme } = useCurrentEuiTheme();
+
+  function onBrushEnd({ x }: XYBrushArea) {
```

Comment:
consider wrapping functions in `useCallback` to avoid new instance on re-render

Reply:
Changed in ac22c8ce86162fd9d4db4687850de4ae33873f4b